spyrmsd/graph.py

```diff
@@ -1,42 +1,102 @@
+from spyrmsd import constants
+
+import numpy as np
+import warnings
+
+_available_backends = []
+_current_backend = None
+
+## Backend aliases
+_graph_tool_aliases = ["graph_tool", "graphtool", "graph-tool", "graph tool", "gt"]
+_networkx_aliases = ["networkx", "nx"]
+
+## Construct the alias dictionary
+_alias_backendDict = {}
+for alias in _graph_tool_aliases:
+    _alias_backendDict[alias.lower()] = "graph-tool"
+for alias in _networkx_aliases:
+    _alias_backendDict[alias.lower()] = "networkx"
+
 try:
     from spyrmsd.graphs.gt import (
-        cycle,
-        graph_from_adjacency_matrix,
-        lattice,
-        match_graphs,
-        num_edges,
-        num_vertices,
-        vertex_property,
+        cycle as gt_cycle,
+        graph_from_adjacency_matrix as gt_graph_from_adjacency_matrix,
+        lattice as gt_lattice,
+        match_graphs as gt_match_graphs,
+        num_edges as gt_num_edges,
+        num_vertices as gt_num_vertices,
+        vertex_property as gt_vertex_property,
     )
-
+    _available_backends.append("graph-tool")
 except ImportError:
-    try:
-        from spyrmsd.graphs.nx import (
-            cycle,
-            graph_from_adjacency_matrix,
-            lattice,
-            match_graphs,
-            num_edges,
-            num_vertices,
-            vertex_property,
-        )
-    except ImportError:
-        raise ImportError("graph_tool or NetworkX libraries not found.")
-
-__all__ = [
-    "graph_from_adjacency_matrix",
-    "match_graphs",
-    "vertex_property",
-    "num_vertices",
-    "num_edges",
-    "lattice",
-    "cycle",
-    "adjacency_matrix_from_atomic_coordinates",
-]
-
-import numpy as np
-
-from spyrmsd import constants
+    warnings.warn("The graph-tool backend does not seem to be installed.", stacklevel=2)
+    
+try:
+   from spyrmsd.graphs.nx import (
+        cycle as nx_cycle,
+        graph_from_adjacency_matrix as nx_graph_from_adjacency_matrix,
+        lattice as nx_lattice,
+        match_graphs as nx_match_graphs,
+        num_edges as nx_num_edges,
+        num_vertices as nx_num_vertices,
+        vertex_property as nx_vertex_property,
+    )
+   _available_backends.append("networkx")
+except ImportError:
+   warnings.warn("The networkx backend does not seem to be installed.", stacklevel=2)
```

Comment:
What is the effect of `stacklevel=2` here?

Reply:
I agree this is a weird one.. I also have this with the set_backend method.
Basically, without it, this is the warning I recieved when setting the backend looks like this:
`spyrmsd/spyrmsd/graph.py:70: UserWarning: The backend is already graph-tool
  warnings.warn(f"The backend is already {backend}")
`
So for some reason it's referring to the part where the warning is raised in the graph.py file. I found this odd, and after some googling I found the stacklevel argument. After setting the stacklevel=2, this is the warning that's produced: 
`customtest_backend.py:32: UserWarning: The backend is already graph-tool
  graph.set_backend("GT")
`
Now it refers to the part of the user's code in that actually caused the warning to happen. I found this more sensible, but I agree it's peculiar. I can remove all instances where I used stacklevel=2 and remove that part if it's more suitable.